js/src/components/paid-ads/ads-campaign/ads-campaign.js

```diff
@@ -0,0 +1,182 @@
+ * @param {boolean} [props.isOnboardingFlow=false] Whether this component is used in onboarding flow.
+ * @param {'create-ads'|'edit-ads'|'setup-ads'} props.trackingContext A context indicating which page this component is used on. This will be the value of `context` in the track event properties.
+ */
+export default function AdsCampaign( {
+	campaign,
+	headerTitle,
+	headerDescription,
+	onContinue,
+	onSkip = noop,
+	hasError = false,
+	isOnboardingFlow = false,
+	trackingContext,
+} ) {
+	const formContext = useAdaptiveFormContext();
+	const { data: countryCodes } = useTargetAudienceFinalCountryCodes();
+	const { isValidForm, setValue } = formContext;
+	const [ completing, setCompleting ] = useState( null );
+	const [ paidAds, setPaidAds ] = useState( {} );
+
+	useEffect( () => {
+		if ( hasError ) {
+			setCompleting( null );
+		}
+	}, [ hasError ] );
+
+	const handleOnStatesReceived = ( paidAdsValues ) => {
+		setPaidAds( paidAdsValues );
+
+		const { amount } = paidAdsValues;
+		setValue( 'amount', amount );
+	};
+
+	const handleSkipCreateAds = () => {
+		setCompleting( ACTION_SKIP );
+
+		onSkip( paidAds );
+	};
+
+	const handleCompleteClick = ( event ) => {
+		setCompleting( event.target.dataset.action );
+
+		onContinue( paidAds );
+	};
+
+	// The status check of Google Ads account connection is included in `paidAds.isReady`,
+	// because when there is no connected account, it will disable the budget section and set the `amount` to `undefined`.
+	const disabledComplete =
+		completing === ACTION_SKIP || ! paidAds.isReady || ! isValidForm;
+
+	let continueButtonProps = {
+		text: __( 'Continue', 'google-listings-and-ads' ),
+	};
+
+	if ( isOnboardingFlow ) {
+		continueButtonProps = {
+			'data-action': ACTION_COMPLETE,
+			text: __( 'Complete setup', 'google-listings-and-ads' ),
+			eventName: 'gla_onboarding_complete_with_paid_ads_button_click',
+			eventProps: {
+				budget: paidAds.amount,
+				audiences: countryCodes?.join( ',' ),
+			},
+		};
+	}
+
+	const description =
+		headerDescription ||
+		createInterpolateElement(
+			__(
+				'Paid Performance Max campaigns are automatically optimized for you by Google. <link>See what your ads will look like.</link>',
+				'google-listings-and-ads'
+			),
+			{
+				link: (
+					<AppDocumentationLink
+						context={ trackingContext }
+						linkId="see-what-ads-look-like"
+						href="https://support.google.com/google-ads/answer/6275294"
+					/>
+				),
+			}
+		);
+
+	return (
+		<StepContent>
+			<StepContentHeader
+				title={ headerTitle }
+				description={ description }
+			/>
+
+			{ isOnboardingFlow && <PaidAdsFeaturesSection /> }
+
+			<PaidAdsSetupSections
+				onStatesReceived={ handleOnStatesReceived }
+				campaign={ campaign }
+				countryCodes={ countryCodes }
+				showCampaignPreviewCard={
+					trackingContext === 'setup-ads' ||
+					trackingContext === 'create-ads'
+				}
+			/>
+
+			<FaqsSection />
```

Comment:
This `FaqsSection` component still doesn't include the improvements from #[#2531](https://github.com/woocommerce/google-listings-and-ads/pull/2531). Can we cherry-pick those fixes into this branch as well and make sure this doesn't cause a regression when all is finalized?

Reply:
@joemcgill I noticed the changes were merged to the [feature/2458-streamline-onboarding](https://github.com/woocommerce/google-listings-and-ads/tree/feature/2458-streamline-onboarding) branch. I only merged the campaign creation one. If we want to have those changes, then it's better we merge the feature/2458-streamline-onboarding branch?

Reply:
The way I originally expected us to sequence this is for all of the 2458 work to be merged to develop and then pulled back into this branch to resolve merge conflicts before merging this work as well. However, we want to hold all of this work so it ships together. If we already had PRs set up for these feature branches, I think it would look like:

* feature/2458-onboarding-improvements > develop (not merged until everything is ready)
  * feature/2590-consolidate-google-account-cards > feature/2458
  * feature/2459-campaign-creation > feature/2458
    * feature/2460-simplify-paid-ads-setup > feature/2459 (these are getting combined)
* feature/2460-google-ads-value-prop (post onboarding) > develop (refreshed after 2458 is merged)

So I think there are two ways of managing this, we could either merge feature/2458 into  feature/2459 and to update/2535 now, or wait and merge feature/2458 into feature/2459 after the feature branch is ready and resolve merge conflicts at that point. Even so, since this moves and replaces some of the changed files, we'll have to be careful about applying those changes here during that merge. I think either strategy is fine.

Reply:
@joemcgill 
The best way forward would be to merge #2458 into #2459 and then we merge everything in 2535.
I have a draft PR set up for it:
https://github.com/woocommerce/google-listings-and-ads/pull/2630

I have another PR ready to merge the updated 2459 into 2535 which we can review afterwards:
https://github.com/woocommerce/google-listings-and-ads/pull/2627

So right now, I think if we can review and merge https://github.com/woocommerce/google-listings-and-ads/pull/2630 and then the campaign creation branch would be the target branch for all future PRs.
Let me know what you think.

Reply:
This sounds right to me. To make things a bit clearer, I'll create draft PRs for the feature branches because 2590 and 2459 should both me opened against 2458, that way we'll be able to see the conflicts that will need to be resolved in an upstream branch.

Reply:
@asvinb I believe I've gotten all the feature branches updated and set up so that this should be ready to finish up. I've created the following draft PRs for the feature branches:

- #2631
- #2632
- #2634

In doing so, I realized that we somehow had a duplicate feature branch for #2509 with a typo in the name (feature/2590... instead of feature/2509...) so I've updating all existing PRs to use the correct feature branch and deleted feature/2590...

I've merged all of the upstream changes from your merge conflict PRs here as well and fixed up an additional merge conflict that remained, so we should be in good shape now, but let me know if you have any issues.